patterns/singleton/singleton.kt

```diff
@@ -0,0 +1,35 @@
+/* Kotlin handles the instantiation 'lazily' (which means it's not created until it's actually needed) and ensures it's thread-safe (which means that it can be safely executed by multiple threads simultaneously without duplicating the instance)
+
+This singleton example creates a private and mutable empty list for names, a method for adding names to the list, and a method for getting the names on the list. 
+
+Coded with help from https://www.kodeco.com/23623842-object-in-kotlin-and-the-singleton-pattern and https://kotlinlang.org/docs/home.html */
+
+object NamesDatabase {
+    private val namesList = mutableListOf<String>()
+
+// Add a name to the list
+    fun addName(name: String) {
+        namesList.add(name)
+    }
+
+// Get a name from the list with index
+    fun getName(index: Int): String = namesList[index]
```

Comment:
This is a bit of an odd example, since it's difficult to imagine wanting to get a name from a database based on the index. Perhaps modify this example to use the [contains](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/contains.html) function?

Reply:
Modified example to use contains; makes a lot more sense.